components/checkbox/CheckboxGroup.razor.cs

```diff
@@ -35,15 +57,49 @@ internal void AddItem(Checkbox checkbox)
         {
             this._checkboxItems ??= new List<Checkbox>();
             this._checkboxItems?.Add(checkbox);
+
+            checkbox.IsFromOptions = IsIsFromOptions(checkbox);
+            if (!checkbox.IsFromOptions)
+            {
+                checkbox.SetValue(_selectedValues.Contains(checkbox.Label));
+                if (_indexConstructedOptionsOffset == -1)
+                    _indexConstructedOptionsOffset = _checkboxItems.Count - 1;
+            }
+            else if (checkbox.IsFromOptions && _indexSetOptionsOffset == -1)
+                _indexSetOptionsOffset = _checkboxItems.Count - 1;
+        }
+
+        private bool IsIsFromOptions(Checkbox checkbox)
+        {
+            if (Options.Value is not null)
+            {
+                if (ChildContent is not null)
+                {
+                    bool result = false;
+                    Options.Switch(
+                        opt => result = opt.Any(o => o.Label.Equals(checkbox.Label)),
+                        arr => result = arr.Contains(checkbox.Label));
+                    return result;
```

Comment:
You maybe could use the `result = Options.Match()` .

Reply:
Ha. Forgot about that 😉